include/drivers/counter.h

```diff
@@ -460,6 +501,72 @@ static inline u32_t z_impl_counter_get_max_relative_alarm(struct device *dev)
 	return api->get_max_relative_alarm(dev);
 }
 
+/**
+ * @brief Set guard period in counter ticks.
+ *
+ * Setting non-zero guard period enables detection of setting absolute alarm
+ * too late. It limits how far in the future absolute alarm can be set.
```

Comment:
Perhaps it would be good to explicitly state that setting the guard period to 0 disables the detection, or is it obvious?

Reply:
`Setting non-zero guard period enables detection of setting absolute alarm too late.` i feel like stating that setting 0 means disabling is redundant.